sdk/identity/identity/src/credentials/clientCertificateCredential.ts

```diff
@@ -9,7 +9,33 @@ import { trace } from "../util/tracing";
 import { MsalFlow } from "../msal/flows";
 import { ClientCertificateCredentialOptions } from "./clientCertificateCredentialOptions";
 
-const logger = credentialLogger("ClientCertificateCredential");
+const credentialName = "ClientCertificateCredential";
+const logger = credentialLogger(credentialName);
+
+/**
+ * Required configuration options for the {@link ClientCertificateCredential}, with either the string contents of a PEM certificate, or the path to a PEM certificate.
+ */
+export type ClientCertificateCredentialPEMConfiguration =
+  | {
+      /**
+       * The PEM-encoded public/private key certificate on the filesystem.
+       */
+      certificate: string;
+      /**
+       * The PEM-encoded public/private key certificate on the filesystem     should not be provided if `certificate` is provided.
+       */
+      certificatePath?: never;
```

Comment:
this is clever to avoid folks providing both on accident, at least in TS

Reply:
Nitpick, I think `certificatePath?: undefined;` is slightly better. It makes it clear that if you access this value you will get `undefined`, which is assignable only to other `undefined` things, whereas `never` is assignable to any constraint.

Reply:
The only thing that I find weird is that if I wanted to use `undefined` here I also need to put the question mark, so `certificatePath?: undefined`, which seems redundant, but it’s necessary for the exclusive-or type to work. Perhaps I’m missing something 🤔

Reply:
I think what's weird is because we want strictly `{ a: string } | { b: string }`, but because `{a: "foo", b: "bar" }` satisfies both of them using TS rules, we have to tell it `{ a: string, b?: undefined } | { a?: undefined, b: string }` because we want to support `{a: "foo"}` and `{b: "bar"}` (the other prop is not present at all, hence the question mark) but we also want the compiler to complain if the key is present and set to anything *except* undefined (like a string, which would create the confusion.)

Reply:
@xirzec I think you have captured what’s weird! I didn’t quite got the suggestion though 🤔